core/src/main/python/wlsdeploy/tool/discover/discoverer.py

```diff
@@ -631,6 +636,24 @@ def wlst_cd(self, path, location):
                             method_name=_method_name)
         return result
 
+    def _inspect_artificial_folder_name(self, folder_name, location):
+        """
+        Perform any special handling for the folder or folder names.
+        :param location: current context of location
+        :return: Original name or processed name value
+        """
+        return self._inspect_security_folder_name(folder_name, location)
+
+    def _inspect_security_folder_name(self, folder_name, location):
+        # This is so clunky. My alias definition change was rejected.
+        if (not self._weblogic_helper.is_version_in_12c()) and self._wlst_mode == WlstModes.OFFLINE and \
+                'SecurityConfiguration' in location.get_folder_path() and 'Provider' == folder_name:
```

Comment:
The location check should be in aliases, maybe is_security_provider(location)?  Rather than checking folder path, the implementation could be is_artificial_type_folder() for now, and expand if other artificial types turn up.